scout/adapter/mongo/hgnc.py

```diff
@@ -300,17 +287,21 @@ def hgncsymbol_to_gene(self, build="37", genes=None):
         LOG.info("All genes fetched")
         return hgnc_dict
 
-    def gene_by_symbol_or_aliases(self, symbol, build="37"):
+    def gene_by_symbol_or_aliases(self, symbol, build=None):
         """Return an iterable with only one gene when gene with a given symbol if found
            or a cursor with genes where the provided symbol is among the aliases.
         Args:
             symbol(str)
-            build(str)
+            build(str or None)
 
         Returns:
             res(list or pymongo.Cursor(dict)): return a list with one gene or a cursor with several gene dictionaries
         """
-        res = self.hgnc_collection.find_one({"hgnc_symbol": symbol, "build": str(build)})
+        query = {"hgnc_symbol": symbol}
+        if build:
+            query["build"] = str(build)
+
+        res = self.hgnc_collection.find_one(query)
```

Comment:
Question/commet:

Why do the functions `gene_by_symbol_or_aliases` and `hgnc_gene_caption` have different approach to setting `query["build"]`?

```
        if build:
            query["build"] = str(build)
```
vs.
```
        if build in ["37", "38"]:
            query["build"] = build
```

Reply:
They could have the same approach, but I think it's because the second is called after the first, so you know that "37" and "38" are strings. If in the second code build is None, then it's not collected